File: watcher-backend/app/schemas/alertas.py
"""
Schemas for Alertas endpoints
"""

from datetime import datetime

from pydantic import BaseModel


class AlertaBase(BaseModel):
    tipo_alerta: str
    nivel_severidad: str
    organismo: str
    programa: str | None = None
    titulo: str
    descripcion: str
    valor_detectado: float | None = None
    valor_esperado: float | None = None
    porcentaje_desvio: float | None = None

class AlertaCreate(AlertaBase):
    boletin_id: int | None = None
    ejecucion_id: int | None = None
    acciones_sugeridas: dict | None = None

class AlertaUpdate(BaseModel):
    estado: str | None = None
    observaciones_revision: str | None = None

class AlertaResponse(AlertaBase):
    id: int
    fecha_deteccion: datetime
    estado: str
    created_at: datetime
    acciones_sugeridas: dict | None = None

    class Config:
        from_attributes = True

class AlertasListResponse(BaseModel):
    alertas: list[AlertaResponse]
    total: int
    page: int
    page_size: int

class AlertasStatsResponse(BaseModel):
    total: int
    criticas: int
    altas: int
    medias: int
    bajas: int
    activas: int
    revisadas: int
    por_tipo: dict
    por_organismo: dict

File: watcher-frontend/src/types/alertas.ts
/**
 * Types for Alertas (Alerts)
 */

export interface Alerta {
  id: number
  tipo_alerta: string
  nivel_severidad: "critica" | "alta" | "media" | "baja"
  organismo: string
  programa?: string | null
  titulo: string
  descripcion: string
  valor_detectado?: number | null
  valor_esperado?: number | null
  porcentaje_desvio?: number | null
  fecha_deteccion: string
  estado: "activa" | "revisada" | "descartada"
  created_at: string
  acciones_sugeridas?: Record<string, unknown> | null
}

export interface AlertasListResponse {
  alertas: Alerta[]
  total: number
  page: number
  page_size: number
}

export interface AlertasStats {
  total: number
  criticas: number
  altas: number
  medias: number
  bajas: number
  activas: number
  revisadas: number
  por_tipo: Record<string, number>
  por_organismo: Record<string, number>
}

export interface AlertaUpdate {
  estado?: "activa" | "revisada" | "descartada"
  observaciones_revision?: string
}

export interface AlertasFilters {
  nivel_severidad?: string
  tipo_alerta?: string
  organismo?: string
  estado?: string
}
